src/plugins/discover/public/application/apps/main/components/doc_table/components/table_row.tsx

```diff
@@ -0,0 +1,201 @@
+
+    return <div className="truncate-by-height">{fieldElement}</div>;
+  };
+  const inlineFilter = useCallback(
+    (column: string, type: '+' | '-') => {
+      const field = indexPattern.fields.getByName(column);
+      filter(field!, flattenedRow[column], type);
+    },
+    [filter, flattenedRow, indexPattern.fields]
+  );
+
+  const getContextAppHref = () => {
+    return getContextUrl(row._id, indexPattern.id!, columns, filterManager, addBasePath);
+  };
+
+  const getSingleDocHref = () => {
+    return addBasePath(getSingleDocUrl(indexPattern.id!, row._index, row._id));
+  };
+
+  const rowCells = [
+    <td className="kbnDocTableCell__toggleDetails" key="toggleDetailsCell">
+      <EuiButtonEmpty
+        onClick={toggleRow}
+        size="xs"
+        aria-expanded={open}
+        aria-label={i18n.translate('discover.docTable.tableRow.toggleRowDetailsButtonAriaLabel', {
+          defaultMessage: 'Toggle row details',
+        })}
+        data-test-subj="docTableExpandToggleColumn"
+      >
+        {open && <EuiIcon type="arrowDown" color="text" size="s" />}
+        {!open && <EuiIcon type="arrowRight" color="text" size="s" />}
+      </EuiButtonEmpty>
+    </td>,
+  ];
+
+  if (indexPattern.timeFieldName && !hideTimeColumn) {
+    rowCells.push(
+      <TableCell
+        key={indexPattern.timeFieldName}
+        timefield={true}
+        formatted={displayField(indexPattern.timeFieldName)}
+        filterable={Boolean(mapping(indexPattern.timeFieldName)?.filterable && filter)}
+        column={indexPattern.timeFieldName}
+        inlineFilter={inlineFilter}
+      />
+    );
+  }
+
+  if (columns.length === 0 && useNewFieldsApi) {
+    const formatted = formatRow(row, indexPattern);
+
+    rowCells.push(
+      <TableCell
+        key="__document__"
+        timefield={false}
+        sourcefield={true}
+        formatted={formatted}
+        filterable={false}
+        column="__document__"
+        inlineFilter={inlineFilter}
+      />
+    );
+  } else {
+    columns.forEach(function (column: string) {
+      // when useNewFieldsApi is true, addressing to the fields property is safe
+      if (useNewFieldsApi && !mapping(column) && !row.fields![column]) {
+        const innerColumns = Object.fromEntries(
+          Object.entries(row.fields!).filter(([key]) => {
+            return key.indexOf(`${column}.`) === 0;
+          })
+        );
+
+        rowCells.push(
+          <TableCell
+            key={column}
+            timefield={false}
+            sourcefield={true}
+            formatted={formatTopLevelObject(row, innerColumns, indexPattern)}
+            filterable={false}
+            column={column}
+            inlineFilter={inlineFilter}
+          />
+        );
+      } else {
+        const isFilterable = Boolean(mapping(column)?.filterable && filter);
+        rowCells.push(
+          <TableCell
+            key={column}
+            timefield={false}
+            sourcefield={column === '_source'}
```

Comment:
won't this be called even if `useNewFieldsApi` is true?

Reply:
Discussed offline